v1/frame.js

```diff
@@ -98,6 +98,19 @@ TChannelFrame.prototype.checksum = function () {
     return csum;
 };
 
+TChannelFrame.prototype.verifyChecksum = function () {
+    var self = this;
+    var err = null;
+    var actual = self.checksum();
+    var expected = self.header.csum;
+    if (expected !== actual) {
+        err = new Error('tchannel checksum validation failed');
```

Comment:
we should set something like `err.type = 'tchannel.frame.failed-checksum'`.

In the future we may want to handle checksum failures differently from other parser failures.